dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/MetricAdapter.java

```diff
@@ -249,42 +248,47 @@ private static void addGaugeData(
   /**
    * This function first transforms the given labels to dimensions using {@link
    * #convertLabelsToDimensions(Labels)} and then adds static labels that are stored in the
-   * singleton instance, if there are any.
+   * singleton instance, if there are any. If two labels have the same key, the last added label is
+   * retained. Constant dimensions such as tags and OneAgent metadata labels are added last.
    *
    * @param labels the labels to be transformed by {@link #convertLabelsToDimensions}.
-   * @return A list of {@link Dimension} objects, that will be serialized.
+   * @return A list of {@link Dimension} objects to be serialized.
    */
   private static List<Dimension> getCombinedDimensions(Labels labels) {
-    List<Dimension> dynamicDimensions = convertLabelsToDimensions(labels);
+    Set<Dimension> dynamicDimensions = convertLabelsToDimensions(labels);
 
     if (getInstance().constantDimensions != null) {
       dynamicDimensions.addAll(getInstance().constantDimensions);
     }
 
-    return Collections.unmodifiableList(dynamicDimensions);
+    return Collections.unmodifiableList(new ArrayList<>(dynamicDimensions));
   }
 
   /**
    * Converts a Map of key-value labels to a List of MINT compatible Dimensions.
    *
    * @param labels are the labels generated by OT.
    * @return all sanitized Dimensions. List is empty, if a key-value pair doesn't match the MINT
-   *     requirements
+   *     requirements. The last specified label for each key will be kept, if multiple labels with
+   *     the same key exists.
    */
-  private static List<Dimension> convertLabelsToDimensions(Labels labels)
+  private static Set<Dimension> convertLabelsToDimensions(Labels labels)
       throws DynatraceExporterException {
-    final List<Dimension> dimensions = new ArrayList<>();
+    // LinkedHashSet will retain the order in which the elements are added.
+    final Set<Dimension> dimensions = new LinkedHashSet<>();
     labels.forEach(
-        new BiConsumer<String, String>() {
-          @Override
-          public void accept(String k, String v) {
+        (String k, String v) -> {
+          try {
             dimensions.add(toMintDimension(k, v));
+          } catch (DynatraceExporterException dee) {
+            logger.warning(
+                String.format(
+                    "Could not transform '%s/%s' to MINT dimension: %s", k, v, dee.getMessage()));
```

Comment:
```suggestion
                    "Could not transform OTel label '%s'->'%s' to Dynatrace dimension: %s", k, v, dee.getMessage()));
```
MINT is no customer-facing terminology, it's just the internal abbreviation for the Metrics ingest API.